<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PreferencesDialog</class>
 <widget class="QDialog" name="PreferencesDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>727</width>
    <height>347</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>0</width>
    <height>0</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Tulip preferences</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../../../../library/tulip-gui/resources/TulipGUIResource.qrc">
    <normaloff>:/tulip/gui/icons/logo32x32.png</normaloff>:/tulip/gui/icons/logo32x32.png</iconset>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_5">
   <item>
    <widget class="QFrame" name="horizontalFrame_2">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_4">
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="tab">
      <attribute name="title">
       <string>Automatic actions</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <widget class="QCheckBox" name="displayDefaultViews">
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;When creating a new graph, importing or loading a graph file, a &lt;span style=&quot; font-weight:600;&quot;&gt;Node Link diagram&lt;/span&gt; and a &lt;span style=&quot; font-weight:600;&quot;&gt;Spreadsheet&lt;/span&gt; views will be automatically created.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="text">
            <string>Display Node Link diagram and Spreadsheet views at create/import/load time</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="aspectRatioCheck">
           <property name="toolTip">
            <string>&lt;html&gt;&lt;body&gt;&lt;p&gt;After applying a layout algorithm, the graph layout will be automatically scaled to get an aspect ratio of 1.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="text">
            <string>Adjust aspect ratio when changing layout</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="centerViewCheck">
           <property name="toolTip">
            <string>&lt;html&gt;&lt;body&gt;&lt;p&gt;After computing a new layout or applying a general algorithm, the graph views will be automatically centered.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="text">
            <string>Center views when changing layout or applying general algorithms</string>
           </property>
           <property name="checked">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="viewOrthoCheck">
           <property name="toolTip">
            <string>&lt;html&gt;&lt;body&gt;&lt;p&gt;If checked, the 3D views will use an orthogonal projection; a central projection if not.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="text">
            <string>Use Orthogonal projection by default (Central projection if not)</string>
           </property>
           <property name="checked">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="colorMappingCheck">
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;After a metric computation, the graph nodes will be automatically colorized according to the metric associated values, using the color scale defined as value of the &amp;quot;color scale&amp;quot; parameter of the &lt;span style=&quot; font-weight:600;&quot;&gt;Color Mapping&lt;/span&gt; plugin.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="text">
            <string>Map colors when computing a metric</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="resultPropertyStoredCheck">
           <property name="toolTip">
            <string>&lt;html&gt;&lt;body&gt;&lt;p&gt;The contents of the result property of an algorithm (viewMetric for example) will be stored in a new property automatically named using the values of the parameters of the algorithm.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="text">
            <string>Store computed result property of an algorithm in a dedicated property</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="usetlpbformat">
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Use the Tulip binary graph file format (tlpb) instead of the tlp plain text file format. The use of the tlpb format makes loading/saving of Tulip project file much faster but you will lose compatibility with older version of Tulip.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="text">
            <string>Use Tulip binary graph file format when saving a Tulip project</string>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_7">
           <item>
            <widget class="QCheckBox" name="randomSeedCheck">
             <property name="toolTip">
              <string>&lt;html&gt;&lt;body&gt;&lt;p&gt;When checked, each random sequence will be initialized with the given value.
If not, a new random seed will be set to initialize each new random sequence.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
             </property>
             <property name="text">
              <string>Give seed for random sequence</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="randomSeedEdit">
             <property name="toolTip">
              <string>The seed value must be a positive integer</string>
             </property>
             <property name="text">
              <string>0</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_4">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_8">
           <item>
            <widget class="QLabel" name="logLabel">
             <property name="toolTip">
              <string>&lt;p&gt;Choose whether you want to log every interactive call to an algorithm, an export or import plugin.&lt;br/&gt;The execution time can be logged too.&lt;/p&gt;</string>
             </property>
             <property name="styleSheet">
              <string notr="true">QLabel  {font-style: normal }</string>
             </property>
             <property name="text">
              <string>Log every interactive call to a plugin:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="logCombo">
             <property name="toolTip">
              <string>&lt;p&gt;Choose whether you want to log every interactive call to an algorithm, an export or import plugin.&lt;br/&gt;The execution time can be logged too.&lt;/p&gt;</string>
             </property>
             <item>
              <property name="text">
               <string>No</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Yes</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Yes, with the execution time</string>
              </property>
             </item>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_5">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_2">
      <attribute name="title">
       <string>Drawing defaults</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <layout class="QVBoxLayout" name="renderingFrameContents">
         <property name="spacing">
          <number>10</number>
         </property>
         <item>
          <widget class="QTableWidget" name="graphDefaultsTable">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="styleSheet">
            <string notr="true">QAbstractItemView {
border: 0px;
}
QHeaderView::section {
height: 30px;
font: bold 12px;
color: black;
border: 0px;
}
QTableView {
font: 12px;
}
</string>
           </property>
           <property name="frameShape">
            <enum>QFrame::NoFrame</enum>
           </property>
           <property name="editTriggers">
            <set>QAbstractItemView::AllEditTriggers</set>
           </property>
           <property name="showGrid">
            <bool>false</bool>
           </property>
           <property name="cornerButtonEnabled">
            <bool>false</bool>
           </property>
           <attribute name="horizontalHeaderDefaultSectionSize">
            <number>200</number>
           </attribute>
           <attribute name="horizontalHeaderMinimumSectionSize">
            <number>1</number>
           </attribute>
           <row>
            <property name="text">
             <string>New Row</string>
            </property>
           </row>
           <row>
            <property name="text">
             <string>New Row</string>
            </property>
           </row>
           <row>
            <property name="text">
             <string>New Row</string>
            </property>
           </row>
           <row>
            <property name="text">
             <string>New Row</string>
            </property>
           </row>
           <row>
            <property name="text">
             <string>New Row</string>
            </property>
           </row>
           <column>
            <property name="text">
             <string/>
            </property>
           </column>
           <column>
            <property name="text">
             <string>Nodes</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>Edges</string>
            </property>
           </column>
           <item row="0" column="0">
            <property name="text">
             <string>Default color</string>
            </property>
            <property name="font">
             <font>
              <pointsize>8</pointsize>
              <weight>75</weight>
              <bold>true</bold>
             </font>
            </property>
           </item>
           <item row="1" column="0">
            <property name="text">
             <string>Default size</string>
            </property>
            <property name="font">
             <font>
              <pointsize>8</pointsize>
              <weight>75</weight>
              <bold>true</bold>
             </font>
            </property>
           </item>
           <item row="2" column="0">
            <property name="text">
             <string>Default shape</string>
            </property>
            <property name="font">
             <font>
              <pointsize>8</pointsize>
              <weight>75</weight>
              <bold>true</bold>
             </font>
            </property>
           </item>
           <item row="3" column="0">
            <property name="text">
             <string>Selection color (nodes = edges)</string>
            </property>
            <property name="font">
             <font>
              <pointsize>8</pointsize>
              <weight>75</weight>
              <bold>true</bold>
             </font>
            </property>
           </item>
           <item row="4" column="0">
            <property name="text">
             <string>Default label color (nodes = edges)</string>
            </property>
            <property name="font">
             <font>
              <pointsize>8</pointsize>
              <weight>75</weight>
              <bold>true</bold>
             </font>
            </property>
           </item>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="drawingHorizontalLayout">
         <property name="spacing">
          <number>0</number>
         </property>
         <item>
          <widget class="QCheckBox" name="applyDrawingDefaultsCheck">
           <property name="toolTip">
            <string>&lt;p&gt;If checked, when clicking &quot;OK&quot;, &lt;br/&gt;the drawing defaults changes will be applied to the graphs currenty loaded.&lt;/p&gt;</string>
           </property>
           <property name="text">
            <string>Apply drawing defaults changes to loaded graphs</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_6">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QPushButton" name="resetAllDrawingDefaultsButton">
           <property name="toolTip">
            <string>&lt;p&gt;Reset all drawing defaults to their Tulip predefined values.&lt;br/&gt;You can also reset row specific default values in using a contextual menu displayed by a left click on the title of the corresponding row.&lt;/p&gt;</string>
           </property>
           <property name="text">
            <string>Reset all defaults</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_3">
      <attribute name="title">
       <string>Proxy settings</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <layout class="QVBoxLayout" name="networkFrameContents">
         <property name="spacing">
          <number>0</number>
         </property>
         <item>
          <widget class="QCheckBox" name="proxyCheck">
           <property name="styleSheet">
            <string notr="true">font: italic;</string>
           </property>
           <property name="text">
            <string>Enable application proxy</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QFrame" name="networkFrame1">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <layout class="QHBoxLayout" name="horizontalLayout">
            <item>
             <widget class="QLabel" name="labeltype">
              <property name="text">
               <string>Type:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="proxyType">
              <item>
               <property name="text">
                <string>SOCKS 5</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>HTTP</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Caching-only HTTP</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Caching FTP</string>
               </property>
              </item>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QFrame" name="networkFrame2">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <layout class="QHBoxLayout" name="horizontalLayout_3">
            <item>
             <widget class="QLabel" name="labelhost">
              <property name="text">
               <string>Host:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="proxyAddr">
              <property name="placeholderText">
               <string>address</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_2">
              <property name="text">
               <string>:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="proxyPort">
              <property name="maximum">
               <number>65535</number>
              </property>
              <property name="value">
               <number>3128</number>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer_2">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QFrame" name="networkFrame3">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <layout class="QHBoxLayout" name="horizontalLayout_6">
            <item>
             <widget class="QCheckBox" name="proxyAuthCheck">
              <property name="text">
               <string>Use authentication</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="proxyUser">
              <property name="enabled">
               <bool>false</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="proxyPassword">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="echoMode">
               <enum>QLineEdit::Password</enum>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer_3">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="Line" name="line_2">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="../../../../library/tulip-gui/resources/TulipGUIResource.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>proxyCheck</sender>
   <signal>clicked(bool)</signal>
   <receiver>networkFrame1</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>538</x>
     <y>107</y>
    </hint>
    <hint type="destinationlabel">
     <x>538</x>
     <y>139</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>proxyCheck</sender>
   <signal>clicked(bool)</signal>
   <receiver>networkFrame3</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>538</x>
     <y>107</y>
    </hint>
    <hint type="destinationlabel">
     <x>538</x>
     <y>205</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>proxyAuthCheck</sender>
   <signal>clicked(bool)</signal>
   <receiver>proxyUser</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>212</x>
     <y>190</y>
    </hint>
    <hint type="destinationlabel">
     <x>397</x>
     <y>190</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>proxyAuthCheck</sender>
   <signal>clicked(bool)</signal>
   <receiver>proxyPassword</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>212</x>
     <y>190</y>
    </hint>
    <hint type="destinationlabel">
     <x>614</x>
     <y>190</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>PreferencesDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>478</x>
     <y>446</y>
    </hint>
    <hint type="destinationlabel">
     <x>478</x>
     <y>235</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>PreferencesDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>478</x>
     <y>446</y>
    </hint>
    <hint type="destinationlabel">
     <x>478</x>
     <y>235</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>proxyCheck</sender>
   <signal>clicked(bool)</signal>
   <receiver>networkFrame2</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>538</x>
     <y>107</y>
    </hint>
    <hint type="destinationlabel">
     <x>538</x>
     <y>173</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>randomSeedCheck</sender>
   <signal>clicked(bool)</signal>
   <receiver>randomSeedEdit</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>158</x>
     <y>336</y>
    </hint>
    <hint type="destinationlabel">
     <x>405</x>
     <y>336</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>settingsChanged()</slot>
  <slot>applySettings()</slot>
 </slots>
</ui>
